test/spec/features/context-pad/ContextPadSpec.js

```diff
@@ -654,10 +653,20 @@ describe('features/context-pad', function() {
         });
       }
 
+      function remove(elements) {
+        getDiagramJS().invoke(function(canvas) {
+          (isArray(elements) ? elements : [ elements ]).forEach(function(element) {
+            canvas.removeShape(element);
+          });
+        });
+
+        change(elements);
```

Comment:
Do we have this behavior covered in `ChangeSupport` specs?

Reply:
You mean that `elements.changed` is fired when a shape is removed? Or that it is considered that elements in `elements.changed` could have been removed?

Reply:
`elements.changed` is always fired even when an element was deleted. GraphicsFactory then [updates all parents](https://github.com/bpmn-io/diagram-js/blob/develop/lib/core/GraphicsFactory.js#L171). Since deleted elements don't have parents anymore they're ignored. `shape.changed` and `connection.changed` aren't fired when elements are deleted.

Reply:
We [did handle delete explicitly in the context pad previously](https://github.com/bpmn-io/diagram-js/commit/854f31692ed52a648e948ed411f32510ab730b90#diff-6d674faf358204aeb4b308e49584a5b425dd2b59dcef70ebfb108f894b96db5dL84) but changed it when the multi-element context pad was introduced.

Reply:
Great explaination, thanks a lot. :medal_sports:

Reply:
Great catch (with this PR).